### ISSUE TITLE
QtSizeSliderPreviewWidget accessing value fails test

### ISSUE THREAD

**Issue by melonora** [napari/napari#5769]
## 🐛 Bug
While running test_qt_size_slider_preview_widget_minimum locally I noticed a view issues with the code of QtSizeSliderPreviewWidget. The test right now depends on a default value that is set in a way that is right now not yet clear to me, but at least it is not hardcoded and thus might break if default values differ on a from system to system. On a side note, shouldn't value be a property instead of a bound method?

## To Reproduce

1. Run test_qt_size_slider_preview_widget_minimum 
2. Test failed multiple times on line 99 for me ```assert widget.value() == 20``` widget value was 33 instead.

## Expected behavior

Test pass

**Comment by Czaki** [comment]
I cannot reproduce it locally. 

But it looks like problem is here: 
https://github.com/napari/napari/blob/1da03500da6ad4368cd43ab4e8662f3fe2aa8c60/napari/_qt/widgets/qt_size_preview.py#L114

As this line depends on local system settings. 

So it looks like thisline should be fixed 
https://github.com/napari/napari/blob/1da03500da6ad4368cd43ab4e8662f3fe2aa8c60/napari/_qt/widgets/_tests/test_qt_size_preview.py#L24
like:

```python
if "value" not in kwargs:
    kwargs["value"] = 12
```

**Comment by melonora** [comment]
Yes that is indeed the line causing the problem. Yesterday when the tests failed the value of self.fontMetrics().heights() returned 33. Today the test passes though. I am not certain yet whether your suggested solution would be correct though as from the Qt documentation it is stated that fontMetrics().height() is always equal to ```widget.fontMetrics().ascent()+ widget.fontMetrics().descent()```. Perhaps it would lead to another bug if ```widget.fontMetrics().ascent()+ widget.fontMetrics().descent() > 12```

**Comment by Czaki** [comment]
My fix will disable querying `fontMetrics` as value will be always provided.

**Comment by melonora** [comment]
I think I am missing some of the logic. The value of the slider would indicate the font size of the preview widget. Right now when doing ```widget.setValue(<int>)``` this does not change the value of ```widget.fontMetrics().height()```, rather it sets the font's pixelsize. 

Sorry if I am not clear, could be that I am just missing something

**Comment by Czaki** [comment]
As I understand, this widget is to control the size of the font. But its starting value is inherited from the default font size. And it may warry between machines and may cause test failing when strat on wrong machine.

**Comment by dalthviz** [comment]
Hi, maybe changing the test to be able to handle the default value being dependant on the local font config could be another option? So instead of setting as minimum `20` to test the value is changing to be in the `[minimum, maximum]` range, maybe using `widget.value() + 1` as minimum could make the test more robust?  (I guess the test was done making the suposition that the OS default font value set is `12` or maybe usually less than `20`?).

So instead of:

https://github.com/napari/napari/blob/1da03500da6ad4368cd43ab4e8662f3fe2aa8c60/napari/_qt/widgets/_tests/test_qt_size_preview.py#L97-L102

Something like:

```python
    current_value = widget.value()
    new_minimum = current_value + 1
    widget.setMinimum(new_minimum)
    assert widget.minimum() == new_minimum
    assert widget.value() == new_minimum
    assert widget._slider.minimum() == new_minimum
    assert widget._slider_min_label.text() == str(new_minimum)
    assert widget._lineedit.text() == str(new_minimum)
```
?
